source/mesh_system.c

```diff
@@ -62,6 +70,7 @@ void mesh_system_init(void)
         platform_timer_enable();
         eventOS_scheduler_init();
         trace_init(); // trace system needs to be initialized right after eventOS_scheduler_init
+        set_trace_print_function(trace_print);
```

Comment:
The function pointers are primarily there to adapt the trace output to the system, so I'm sure this is the correct place. The system's default C library "puts" is broken, so the default trace routine doesn't work and we have to provide an alternative.

For system setup, it makes sense that the component calling trace_init() also set up the rest of the trace for the system. Applications using mbed-mesh-api rely on it to do all the setup, as they're potentially not trace-aware at all.

What it looks like we do need though is a way redirect output functions, then **restore the previous**. Tommi's search found 2 instances of people putting in a temporary custom command output function ("screen" and 1 other I don't remember), then putting back "default_command_print_function". I suggest that set_trace_print_function and set_cmd_print_function(?) return the previous function pointer.

However, the search didn't find any examples of people messing with the trace functions, so this seems like it should be fine.